var/spack/repos/builtin/packages/py-psyclone/package.py

```diff
@@ -8,23 +8,28 @@
 
 class PyPsyclone(PythonPackage):
     """
-    Code generation system developed to support domain-specific languages
-    (DSLs) for finite element, finite volume and finite difference codes.
-    Notably, it is used in the LFRic Project from the UK Met Office, and
-    it also supports the GOcean (2D, finite difference) DSL.
+    A source-to-source Fortran compiler designed to programmatically optimise,
+    parallelise and instrument HPC applications (written in Fortran) via user-
+    provided transformation scripts. Additionally, PSyclone supports the
+    development of kernel-based, Fortran-embedded DSLs and is used in the UK
+    Met Office's next-generation modelling system, LFRic.
 
     """
 
     # Links
     homepage = "https://github.com/stfc/PSyclone"
     git = "https://github.com/stfc/PSyclone.git"
-    pypi = "PSyclone/PSyclone-2.5.0.tar.gz"
+    pypi = "PSyclone/PSyclone-3.0.0.tar.gz"
```

Comment:
Given this change, I recommend fixing line 22 to the new case (assuming it will be used going forward) and implementing for `url_for_version` at least until the older versions are removed from the package (after being deprecated). (See https://spack.readthedocs.io/en/latest/packaging_guide.html#version-urls.)

Reply:
For example, you could add an implementation along the lines of the following -- after all of the dependencies (i.e., as the first method in the class):

```
    def url_for_version(self, version):
        name = "psyclone" if version >= Version("3.0.0") else "PSyclone"
        return f"https://files.pythonhosted.org/packages/source/P/PSyclone/{name}-{version}.tar.gz"
``` 

With that implementation I can (re)confirm all of the version sha256s.

Reply:
Ah! Unfortunately I don't know what caused that change.  I've added the `url_for_version` method as you suggest.